src/packagedcode/npm.py

```diff
@@ -169,6 +195,176 @@ def walk_npm(cls, resource, codebase, depth=0):
                 for subchild in cls.walk_npm(child, codebase, depth=depth):
                     yield subchild
 
+    @classmethod
+    def update_dependencies_by_purl(
+        cls,
+        dependencies,
+        scope,
+        dependecies_by_purl,
+        is_runtime=False,
+        is_optional=False,
+        is_resolved=False,
+        is_direct=True,
+    ):
+
+        metadata_deps = ['peerDependenciesMeta', 'dependenciesMeta']
```

Comment:
What are these? You mentioned these are scopes? Any details, link or docs?

Reply:
Added, thanks